### PR TITLE
New hook - integrate_ini_settings (Session.php)

### PR DESCRIPTION
**Pull request by dragomano**
Signed-off-by: Bugo <bugo@dragomano.ru>

### CODE REVIEW

**Comment by albertlast** [comment]
first question should be a mod able to change the php settings?
second is the database already up at this point?
How are programm error catched by mods at this point of time?

**Comment by dragomano** [comment]
The database is not required there, $modSettings is enough :)
For example, `@ini_set('session.use_only_cookies', !empty($modSettings['some_option']));`

**Comment by albertlast** [comment]
did you check how call_integration_hook is working?

**Comment by dragomano** [comment]
Of course.

**Comment by albertlast** [comment]
modSettings gefilled by the database... 
so you need databse.

File: Sources/Session.php [hunk 1]
```diff
@@ -34,6 +34,9 @@ function loadSession()
 	@ini_set('session.use_trans_sid', false);
 	@ini_set('arg_separator.output', '&amp;');
 
+	// Allows mods to change/add PHP settings
+	call_integration_hook('integrate_ini_settings');
```
Comment (Sesquipedalian) [inline Sources/Session.php]: Having a hook at this point in `loadSession()` makes sense, if the hooked code relates to loading sessions. 

This is not, however, a good place for a generic hook to change arbitrary ini settings, so the name of this hook should not suggest that to mod developers. Please change it to `integrate_load_session`. I know it is only cosmetic, but we want to communicate the right idea with the name.

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]

File: Sources/Session.php (hunk 1 above)
Comment (dragomano) [reply Sources/Session.php]: Ok, done. Currently, it is the only way to change such variables (before starting a session). If you try to change them later, there will be a warning:
`: ini_set(): A session is active. You cannot change the session module's ini settings at this time in...`

**Review by Sesquipedalian — APPROVED** [review]

**Pull request merged**